src/detect-engine-address.c

```diff
@@ -533,7 +533,9 @@ int DetectAddressParseString(DetectAddress *dd, const char *str)
         str++;
 
     /* shouldn't see 'any' here */
-    BUG_ON(strcasecmp(str, "any") == 0);
+    if(strcasecmp(str, "any") == 0) {
```

Comment:
I would rather see with that input we end up here, as the comment tells us this should be impossible

Reply:
I understand the comment as "no one should write a rule with any here" rather than it is impossible to have any here (I need some work to regenerate an input for this case)

Reply:
BUG_ON is not meant for user reachable errors, but for actual code bugs.

Reply:
That is the point of my PR :-)
But I now think you mean that there may be a deeper bug...
I will try to find again the reproducer